Source/JavaScriptCore/dfg/DFGOperations.cpp

```diff
@@ -2787,6 +2787,10 @@ JSC_DEFINE_JIT_OPERATION(operationStringReplaceStringString, JSString*, (JSGloba
     JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
     auto scope = DECLARE_THROW_SCOPE(vm);
 
+    if (JSString* result = tryReplaceOneCharUsingString<>(globalObject, stringCell, searchCell, replacementCell))
+        OPERATION_RETURN(scope, result);
+    OPERATION_RETURN_IF_EXCEPTION(scope, nullptr);
```

Comment:
I think you need to do this in,

```
JSString* result = tryReplaceOneCharUsingString<>(globalObject, stringCell, searchCell, replacementCell);
OPERATION_RETURN_IF_EXCEPTION(scope, nullptr);
if (result)
    OPERATION_RETURN(scope, result);
```
to avoid debug assertions.

Reply:
`OPERATION_RETURN` will release the `scope` and then return the result. In that way, we can do less exception checks if `result` is true.